.github/workflows/kong3.0_compatibility.yaml

```diff
@@ -86,8 +88,94 @@ jobs:
       if: ${{ always() }}
       uses: actions/upload-artifact@v3
       with:
-        name: diagnostics-integration-tests-enterprise-postgres
+        name: diagnostics-integration-${{ matrix.database-mode }}-${{ matrix.kong-edtion }}
         path: /tmp/ktf-diag*
         if-no-files-found: ignore
 
-  # e2e-tests: ## TODO: blocked by loading kong image in manifests: https://github.com/Kong/kubernetes-ingress-controller/issues/2884
+    e2e-tests:
+      environment: "Configure ci"
+      runs-on: ubuntu-latest 
+      steps:
+      - name: setup golang
+        uses: actions/setup-go@v3
+        with:
+          go-version: '^1.19'
+
+      - name: cache go modules
+        uses: actions/cache@v3
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-build-codegen-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-build-codegen-
+
+      - name: checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          driver-opts: network=host
+
+      - name: Build and push to local registry
+        id: docker_build
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          load: true
+          file: Dockerfile
+          tags: kong/kubernetes-ingress-controller:ci
+          target: distroless
+
+      - name: run non-enterprise cases of e2e test
+        run: make test.e2e
+        ## run tests without KONG_LICENSE_DATA env to skip enterprise tests.
+        env:
+          TEST_KONG_CONTROLLER_IMAGE_LOAD: kong/kubernetes-ingress-controller:ci
+          TEST_KONG_CONTROLLER_IMAGE_OVERRIDE: kong/kubernetes-ingress-controller:ci
+          TEST_KONG_IMAGE_LOAD: kong/kong:3.0.0-alpha.14-alpine
+          KONG_CLUSTER_VERSION: "v1.24.2"
+          ISTIO_VERSION: "v1.14.1"
+          NCPU: 1
+
+      - name: upload diagnostics for e2e OSS 
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: diagnostics-e2e-tests-oss
+          path: /tmp/ktf-diag*
+          if-no-files-found: ignore
+
+      - name: setup kong enterprise license
+        uses: Kong/kong-license@master
+        id: license
+        with:
+          # PULP_PASSWORD secret is set in "Configure ci" environment
+          password: ${{ secrets.PULP_PASSWORD }}
+
+      - name: run enterprise cases of e2e test
+        run: make test.e2e ## TODO: the config runs tests for both non-enterprise and enterprise tests. Should we run enterprise tests only in this step?
+        ## run tests without KONG_LICENSE_DATA env to skip enterprise tests.
```

Comment:
I'd separate those so that it's easier to figure out what failed when a failure occurs.

And ideally I'd put those into separate jobs so that they can run in parallel and independently of each other.

Reply:
separate what? integration vs e2e or OSS vs enterprise?

Reply:
This runs inside a job called `e2e-tests` (line 95) which runs both OS and enterprise.

I suggest to split those into 2 so that they run in parallel, separated. WDYT?

Reply:
Running them in parallel would cost less time to get the result, so +1 for this.